packages/api/app/auth.js

```diff
@@ -35,29 +50,24 @@ async function ensureAuthorized(req, res, next) {
     return res.boom.unauthorized('Missing token');
   }
 
-  let userName;
-  let accessToken;
-  try {
-    ({ username: userName, accessToken } = verifyJwtToken(jwtToken));
+  if (launchpadProtectedAuth()) {
+    return ensureLaunchpadAPIAuthorized(req, res, next);
+  }
 
-    const userModel = new User();
-    const access = new AccessToken();
+  try {
+    const { username, accessToken } = verifyJwtToken(jwtToken);
 
-    if (!launchpadProtectedAuth()) {
-      // Only verify user if we're not launchpad protected
-      await userModel.get({ userName });
+    if (!(await isAuthorizedOAuthUser(username))) {
+      return res.boom.unauthorized('User not authorized');
     }
+
+    const access = new AccessToken();
     await access.get({ accessToken });
     // Adds additional metadata that authorized endpoints can access.
-    req.authorizedMetadata = { userName };
+    req.authorizedMetadata = { userName: username };
+
     return next();
   } catch (error) {
-    if (launchpadProtectedAuth()
```

Comment:
I think this needs to be implemented this way based on this comment in the CHANGELOG

https://github.com/nasa/cumulus/blame/master/CHANGELOG.md#L387

Reply:
Refactored and ready for re-review.